File: Java/Hussain Sir Java/Fundamentals/Generics/Basics/GenClassTest.java
class SimpleStack<V>{
	
	private Node top;

	class Node{
		V value;
		Node below;

		Node(V val){
			value = val;
			below = top;
		}
	}

	public void push(V val){
		top = new Node(val);
	}

	public V pop(){
		Node node = top;
		top = top.below;
		return node.value;
	}

	public boolean empty(){
		return top == null;
	}
}

class GenClassTest{

	public static void main(String[] args){
		SimpleStack<String> a = new SimpleStack<String>();
		a.push("Monday");
		a.push("Tuesday");
		a.push("Wednesday");
		a.push("Thursday");
		a.push("Friday");
		SimpleStack<String> b = new SimpleStack<String>();
		b.push("June");
		b.push("May");
		b.push("April");
		b.push("March");
		SimpleStack<Interval> c = new SimpleStack<>();
		c.push(new Interval(5, 41));
		c.push(new Interval(6, 32));
		c.push(new Interval(4, 13));
		c.push(new Interval(7, 24));
		c.push(new Interval(3, 55));
		SimpleStack<Object> d = new SimpleStack<>();
		d.push("Sunday");
		d.push(new Interval(2, 30));
		d.push(4.56);
		while(!a.empty())
			System.out.println(a.pop());
		System.out.println("---------------------");
		while(!b.empty())
			System.out.println(b.pop());
		System.out.println("---------------------");
		while(!c.empty())
			System.out.println(c.pop());
		System.out.println("---------------------");
		while(!d.empty())
			System.out.println(d.pop());
	}
}

File: Fundamentals/Generics/Basics/genclasstest.cs
using System;

class SimpleStack<V>
{
	class Node
	{
		internal V value;
		internal Node below;

		internal Node(V val, Node node)
		{
			value = val;
			below = node;
		}
	}

	private Node top;

	public void Push(V value)
	{
		top = new Node(value, top);
	}

	public V Pop()
	{
		Node node = top;
		top = top.below;
		return node.value;
	}

	public bool Empty()
	{
		return top == null;
	}
}

static class Program
{
	public static void Main()
	{
		SimpleStack<string> a = new SimpleStack<string>();
		a.Push("Monday");
		a.Push("Tuesday");
		a.Push("Wednesday");
		a.Push("Thursday");
		a.Push("Friday");

		SimpleStack<string> b = new SimpleStack<string>();
		b.Push("June");
		b.Push("May");
		b.Push("April");
		b.Push("March");

		SimpleStack<Interval> c = new SimpleStack<Interval>();
		c.Push(new Interval(5, 31));
		c.Push(new Interval(3, 42));
		c.Push(new Interval(7, 13));
		c.Push(new Interval(6, 54));
		c.Push(new Interval(4, 25));

		SimpleStack<object> d = new SimpleStack<object>();
		d.Push("Sunday");
		d.Push(new Interval(2, 30));
		d.Push(34.5);

		while(!a.Empty())
			Console.WriteLine(a.Pop());
		Console.WriteLine("---------------------");

		while(!b.Empty())
			Console.WriteLine(b.Pop());
		Console.WriteLine("---------------------");

		while(!c.Empty())
			Console.WriteLine(c.Pop());
		Console.WriteLine("---------------------");

		while(!d.Empty())
			Console.WriteLine(d.Pop());


	}
}
